### ISSUE TITLE
`which` (spawn) test broken under julia-debug

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#32586]
```
Worker 8 failed running test spawn:
Some tests did not pass: 146 passed, 1 failed, 0 errored, 4 broken.spawn: Test Failed at /data/vtjnash/julia/test/spawn.jl:546
  Expression: Sys.which("julia") == realpath(julia_exe)
   Evaluated: "/data/vtjnash/julia/usr/bin/julia" == "/data/vtjnash/julia/usr/bin/julia-debug"
Stacktrace:
 [1] record(::Test.DefaultTestSet, ::Test.Fail) at /data/vtjnash/julia/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:775
 [2] (::getfield(Main, Symbol("##44#50")))() at /data/vtjnash/julia/test/runtests.jl:267
 [3] cd(::getfield(Main, Symbol("##44#50")), ::String) at ./file.jl:104
 [4] top-level scope at /data/vtjnash/julia/test/runtests.jl:61
 [5] include at ./boot.jl:328 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1094
 [7] include(::Module, ::String) at ./Base.jl:31
 [8] exec_options(::Base.JLOptions) at ./client.jl:295
 [9] _start() at ./client.jl:468
```

**Comment by ararslan** [comment]
That seems like it should be easy enough to fix: just make the LHS something like `Sys.which(basename(first(Base.julia_cmd())))`.

**Comment by vtjnash** [comment]
Fixed in #30381
